src/software/proto/message_translation/ssl_referee_test.cpp

```diff
@@ -2,4 +2,157 @@
+                        SSL_Referee_Stage_NORMAL_SECOND_HALF),
+        std::make_tuple(RefboxStage::EXTRA_TIME_BREAK,
+                        SSL_Referee_Stage_EXTRA_TIME_BREAK),
+        std::make_tuple(RefboxStage::EXTRA_FIRST_HALF_PRE,
+                        SSL_Referee_Stage_EXTRA_FIRST_HALF_PRE),
+        std::make_tuple(RefboxStage::EXTRA_FIRST_HALF,
+                        SSL_Referee_Stage_EXTRA_FIRST_HALF),
+        std::make_tuple(RefboxStage::EXTRA_HALF_TIME, SSL_Referee_Stage_EXTRA_HALF_TIME),
+        std::make_tuple(RefboxStage::EXTRA_SECOND_HALF_PRE,
+                        SSL_Referee_Stage_EXTRA_SECOND_HALF_PRE),
+        std::make_tuple(RefboxStage::EXTRA_SECOND_HALF,
+                        SSL_Referee_Stage_EXTRA_SECOND_HALF),
+        std::make_tuple(RefboxStage::PENALTY_SHOOTOUT_BREAK,
+                        SSL_Referee_Stage_PENALTY_SHOOTOUT_BREAK),
+        std::make_tuple(RefboxStage::PENALTY_SHOOTOUT,
+                        SSL_Referee_Stage_PENALTY_SHOOTOUT),
+        std::make_tuple(RefboxStage::POST_GAME, SSL_Referee_Stage_POST_GAME)));
+
+class RefboxGameStateTest
+    : public ::testing::TestWithParam<
+          std::tuple<RefboxGameState, SSL_Referee_Command, TeamColour>>
+{
+};
+
+TEST_P(RefboxGameStateTest, test_refbox_game_state)
+{
+    RefboxGameState expected  = std::get<0>(GetParam());
+    SSL_Referee_Command input = std::get<1>(GetParam());
+    TeamColour team_colour    = std::get<2>(GetParam());
+
+    SSL_Referee ref;
+    ref.set_command(input);
+
+    ASSERT_EQ(expected, createRefboxGameState(ref, team_colour));
+}
+
+INSTANTIATE_TEST_CASE_P(
+    RefboxGameStateTests, RefboxGameStateTest,
+    ::testing::Values(
+        // kickoff
+        std::make_tuple(RefboxGameState::PREPARE_KICKOFF_US,
+                        SSL_Referee_Command_PREPARE_KICKOFF_YELLOW, TeamColour::YELLOW),
+        std::make_tuple(RefboxGameState::PREPARE_KICKOFF_US,
+                        SSL_Referee_Command_PREPARE_KICKOFF_BLUE, TeamColour::BLUE),
+        std::make_tuple(RefboxGameState::PREPARE_KICKOFF_THEM,
+                        SSL_Referee_Command_PREPARE_KICKOFF_BLUE, TeamColour::YELLOW),
+        std::make_tuple(RefboxGameState::PREPARE_KICKOFF_THEM,
+                        SSL_Referee_Command_PREPARE_KICKOFF_YELLOW, TeamColour::BLUE),
+
+        // penalty
+        std::make_tuple(RefboxGameState::PREPARE_PENALTY_US,
+                        SSL_Referee_Command_PREPARE_PENALTY_YELLOW, TeamColour::YELLOW),
+        std::make_tuple(RefboxGameState::PREPARE_PENALTY_US,
+                        SSL_Referee_Command_PREPARE_PENALTY_BLUE, TeamColour::BLUE),
+        std::make_tuple(RefboxGameState::PREPARE_PENALTY_THEM,
+                        SSL_Referee_Command_PREPARE_PENALTY_BLUE, TeamColour::YELLOW),
+        std::make_tuple(RefboxGameState::PREPARE_PENALTY_THEM,
+                        SSL_Referee_Command_PREPARE_PENALTY_YELLOW, TeamColour::BLUE),
```

Comment:
It might be worth looking at the gtest `Combine` to generate a lot of these combos: https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#value-parameterized-tests

Reply:
TIL, this is pretty cool, combine generates a cartesian product of all the values, which will create invalid combinations making testing hard.

I couldn't figure out a way to "filter" the combines but at that point it seem like maybe defining all of the valid combos is the best way to go?

Reply:
Nah that's fine, this is probably a more reasonable solution as-is then.